### PR TITLE
feat(#815): Integrate with secrets manager

### LINKED ISSUE #815: feat: Integrate with AWS Secrets Manager
# Feature Request

Define a new backend class for integrating with the AWS Secrets Manager service.

## User Story

As a cloud developer, I want to use `config-ninja` to read configuration objects stored in AWS Secrets Manager.

### PR DESCRIPTION
**Pull request by bryant-finney**
Closes #815

### CODE REVIEW

File: src/config_ninja/contrib/secretsmanager.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    """The `boto3` client used to communicate with the AWS Secrets Manager service."""
+
+    secret_id: str
+    """The ID of the secret to retrieve"""
+
+    version_id: str | None = None
+
+    def __init__(self, client: SecretsManagerClient, secret_id: str) -> None:
+        """Initialize the backend."""
+        logger.debug("Initialize: %s(client=%s, '%s')", self.__class__.__name__, client, secret_id)
```
[the remark this answers is not in the document]
Comment (bryant-finney) [reply src/config_ninja/contrib/secretsmanager.py:53]: What? `secret_id` is not a secret, stfu

[the remark this answers is not in the document]
Comment (bryant-finney) [reply src/config_ninja/contrib/secretsmanager.py]: More interesting: dc51ad63190ca07323dcfc4e8649b7df3f5b6687 appeased CodeQL, but the value of `secret_id` is still getting logged 🙂

File: src/config_ninja/contrib/secretsmanager.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                version_id = self._retrieve_current_version()
+            except ValueError as exc:
+                logger.warning('%s', exc)
+            else:
+                if version_id and version_id != self.version_id:
+                    yield self.get()
+
+            await asyncio.sleep(interval)
+
+    def _async_doctests(self) -> None:
```
Comment (jcorbettfrank) [inline src/config_ninja/contrib/secretsmanager.py:101]: asked copilot about some things in this file, but this is :fire:

**Review by jcorbettfrank — APPROVED** [review]
Looks great. TIL about doctests :fire:

File: src/config_ninja/contrib/secretsmanager.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcorbettfrank) [reply src/config_ninja/contrib/secretsmanager.py]: :slightly_smiling_face:

**Pull request merged**